### PR TITLE
executor: GROUP_CONCAT(float) is not compatible with mysql

### LINKED ISSUE #23312: GROUP_CONCAT(folat) is not compatible with mysql
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```mysql
CREATE TABLE `t2`  (
  `col1` float GENERATED ALWAYS AS ((`col102` + 10)) STORED NOT NULL,
  `col102` float,
  PRIMARY KEY (`col1`) USING BTREE
)

INSERT INTO `t2` VALUES (DEFAULT, 3.36769e38);
INSERT INTO `t2` VALUES (DEFAULT, 3.18374e38);
```

### 2. What did you expect to see? (Required)
```mysql
mysql> select  GROUP_CONCAT(col1)  from t2;
+-----------------------+
| GROUP_CONCAT(col1)    |
+-----------------------+
| 3.18374e38,3.36769e38 |
+-----------------------+
1 row in set (0.03 sec)

```
### 3. What did you see instead (Required)
```mysql
mysql> select  GROUP_CONCAT(col1)  from t2;
+---------------------------------------------------------------------------------+
| GROUP_CONCAT(col1)                                                              |
+---------------------------------------------------------------------------------+
| 318374000000000000000000000000000000000,336769000000000000000000000000000000000 |
+---------------------------------------------------------------------------------+
1 row in set (0.06 sec)
```
### 4. What is your TiDB version? (Required)
```mysql
Community
Git Commit Hash: 058e52ad7c1b477147dcb933b3f1b3b2be31e998
Git Branch: heads/refs/tags/v4.0.11
UTC Build Time: 2021-02-25 04:33:58
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by Howie59**
### What problem does this PR solve?

Issue Number: close #23312 

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Review by djshow832 — COMMENTED** [review]
Can you add some test cases?

**Comment by guo-shaoge** [comment]
Looks like this case `testIntegrationSuite2.TestEncryptionBuiltin` is failed. Would you address this? @Howie59

File: expression/builtin_cast_vec.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func (b *builtinCastRealAsStringSig) vecEvalString(input *chunk.Chunk, result *c
 			result.AppendNull()
 			continue
 		}
-		res, err = types.ProduceStrWithSpecifiedTp(strconv.FormatFloat(v, 'f', -1, bits), b.tp, sc, false)
+		res, err = types.ProduceStrWithSpecifiedTp(strconv.FormatFloat(v, 'e', -1, bits), b.tp, sc, false)
```
Comment (guo-shaoge) [inline expression/builtin_cast_vec.go:224]: Looks like this change may cause compatible issues.

    drop table if exists t;
    create table t(c1 float(16, 4));
    insert into t1 values(3.1415926535);

Before change, compatible:

    mysql> select concat('a', c1) from t1;
    +-----------------+
    | concat('a', c1) |
    +-----------------+
    | a3.1416         |
    +-----------------+
    1 row in set (0.00 sec)

After changed, not compatible.

    mysql> select concat('a', c1) from t1;
    +-----------------+
    | concat('a', c1) |
    +-----------------+
    | a3.1416e+00     |
    +-----------------+

**Pull request closed without merging**